src/Diagnostics.DataProviders/DataProviders/SupportObserverDataProvider.cs

```diff
@@ -107,14 +107,31 @@ public override async Task<string> GetWebspaceResourceGroupName(string subscript
 
         public override async Task<dynamic> GetSite(string siteName)
         {
-            var response = await GetObserverResource($"sites/{siteName}");
-            var siteObject = JsonConvert.DeserializeObject(response);
-            return siteObject;
+            return await GetSiteInternal(null, siteName, null);
         }
 
         public override async Task<dynamic> GetSite(string stampName, string siteName)
         {
-            var response = await GetObserverResource($"stamps/{stampName}/sites/{siteName}");
+            return await GetSiteInternal(stampName, siteName, null);
+        }
+
+        public override async Task<dynamic> GetSite(string stampName, string siteName, string slotName)
+        {
+            return await GetSiteInternal(stampName, siteName, slotName);
+        }
+
+        private async Task<dynamic> GetSiteInternal(string stampName, string siteName, string slotName)
+        {
+            string path = "";
+
+            if (!string.IsNullOrWhiteSpace(stampName))
+            {
+                path = $"stamps/{stampName}/";
+            }
+
+            path = slotName == null ? path + $"sites/{siteName}" : path + $"sites/{siteName}({slotName})";
```

Comment:
update the check to nullorWhitespace.. you dont want to query 'sitename()'

Reply:
nowadays I search my code for string.IsNullOrEmpty before filing a PR as I just cannot escape that from Shekhar's eyes :-D